### PR TITLE
Allow running in docker

### PR DESCRIPTION
**Pull request by techhazard**
I wanted to be able to run taskwarrior in docker. There was no `Dockerfile` that I could find, so I made one.

The final image is around 100MB in size. 
I was unable to statically compile/link (my gcc-fu isn't good enough :-p). If that's possible, we can reduce probably return it to under 10MB, especially if it can be linked with musl-libc.

The image can be run with the following:
```sh
docker build -t taskserver .
docker run --rm --name taskserver -p 53589:53589 taskserver
```
Taskserver should now be available on port `53589`

I hope you like it :slightly_smiling_face:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  && rm -rf /var/cache/apt/
+
+COPY ./CMakeLists.txt /build/
+COPY ./src /build/src
+COPY ./doc /build/doc
+COPY ./cmake.* /build/
+COPY ./cmake /build/cmake
+
+RUN cmake -DCMAKE_BUILD_TYPE=release . 
+RUN make -j3
```
Comment (jrabbit) [inline Dockerfile:33]: This shouldn't be hardcoded

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    uuid-dev \
+    cmake \
+    gnutls-bin \
+  && rm -rf /var/cache/apt/
+
+COPY ./CMakeLists.txt /build/
+COPY ./src /build/src
+COPY ./doc /build/doc
+COPY ./cmake.* /build/
+COPY ./cmake /build/cmake
```
Comment (jrabbit) [inline Dockerfile:30]: Could we just copy the whole source tree?

**Review by jrabbit — CHANGES_REQUESTED** [review]
Seems ok we need this rebased or we'd need to cherry-pick it

**Comment by pbeckingham** [comment]
380 commits on master? We don't work on master, please resubmit as PR on the latest development branch.

**Comment by hickey** [comment]
For anyone following this PR, I have produced a Docker container for `taskd`. This work was done due to the shutdown of Freecinc and I figured that there would be others looking to bring up their own taskservers to sync between multiple machines. The container weights in just under 20MB. Project is over at https://gitlab.com/wt0f/taskd/

**Pull request closed without merging**